### PR TITLE
feat: send email to the user on ID verification

### PR DESCRIPTION
**Pull request by arjunsinghy96**
Uses edx-ace to send emails on student verification application
rejection and approval.

Related Tickets: [BB-4221]()

#### Testing Instructions

1. On a `juniper` devstack, clone this under `{devstack_folder}/src` (devstack_folder is the one that contains edx-platform, devstack, and the other folders.
1. Get into the LMS shell using `make lms-shell`.
1. Install the app using pip: `pip install -e /edx/src/custom-student-verification`
2. Restart LMS
3. Log into LMS with an unverified user and visit http://localhost:18000/custom-student-verification
4. Upload an image and submit
5. With a staff user logged in visit http://localhost:18000/admin/custom_student_verification/studentverificationrequest/
6. Open the verification request created above, mark it `approved` and save.
7. Verify that an email is sent by checking the latest file in `/edx/src/ace_messages` in `lms-shell`
8. Open the verification request created above again in your browser.
9. Mark the request `rejected` and save.
10. Verify that a rejected email is send by checking the file in `/edx/src/ace_messages`.
11. The above emails can also be seen in `lms-logs`

#### Author Notes
1. The email will be sent every time the verification request is marked `rejected`. This includes just opening a `rejected` request and saving it again from browser.
2. When the request is marked `rejected`, the `approved` records from `ManualVerification` are deleted.
3. The requirements are updated since `edx_ace` is added.

### CODE REVIEW

File: custom_student_verification/templates/custom_student_verification/edx_ace/verificationrejected/email/body.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <br/>
+        {% filter force_escape %}
+          {% blocktrans %}The {{ platform_name }} Team {% endblocktrans %}
+        {% endfilter %}
+      </p>
+
+    </td>
+  </tr>
+</table>
+{% endblock %}
```
Comment (shimulch) [inline custom_student_verification/templates/custom_student_verification/edx_ace/verificationrejected/email/body.html]: Nit :)  Add a New line.

File: custom_student_verification/signals.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ def on_student_verification_request_saved(sender, instance, **kwargs):  # pylint
     """
     if instance.status == 'ACCEPTED':
         approve_user(instance.user, instance.reason)
+    elif instance.status == 'REJECTED':
+        reject_user_verification(instance.user, instance.reason)
```
Comment (shimulch) [inline custom_student_verification/signals.py]: @arjunsinghy96 Can we use a similar naming for both functions? `approve_user` and `reject_user_verification`?

File: requirements/dev.txt [hunk 3]
```diff
@@ -5,87 +5,87 @@
...
+codecov==2.1.11           # via -r requirements/travis.txt
+coverage[toml]==5.5       # via -r requirements/quality.txt, -r requirements/travis.txt, codecov, pytest-cov
+ddt==1.4.2                # via -r requirements/quality.txt
+diff-cover==4.2.3         # via -r requirements/dev.in
 distlib==0.3.1            # via -r requirements/quality.txt, -r requirements/travis.txt, caniusepython3, virtualenv
-django==2.2               # via -c requirements/constraints.txt, -r requirements/quality.txt, code-annotations, djangorestframework, edx-i18n-tools
-djangorestframework==3.11.1  # via -r requirements/quality.txt
+django==2.2               # via -c requirements/constraints.txt, -r requirements/quality.txt, code-annotations, djangorestframework, edx-ace, edx-i18n-tools
+djangorestframework==3.12.4  # via -r requirements/quality.txt
+edx-ace==0.1.17           # via -r requirements/quality.txt
```
Comment (shimulch) [inline requirements/dev.txt:24]: @arjunsinghy96 `edx-ace` should be in `base.in` right? But I don't see that change. Have I missed it?

**Review by shimulch — CHANGES_REQUESTED** [review]
@arjunsinghy96 Thanks a lot for the PR. The PR looks good, just need to address some nits :)

File: requirements/dev.txt (hunk 3 above)
Comment (arjunsinghy96) [reply requirements/dev.txt:24]: @shimulch I have kept `edx-ace` in `test.in` as `LMS` will already have `edx-ace` and these two dependencies might conflict. (Just to be on the safe side)
For this project we need `edx-ace` primarily for testing as the application will not be run stand-alone.

However, please let me know if keeping it in base.in makes more sense.

Comment (shimulch) [reply requirements/dev.txt:24]: @arjunsinghy96 We should keep our core dependencies on the `base.in`. Check the [file](https://github.com/open-craft/custom-student-verification/blob/main/requirements/base.in). If there were any issue with dependency conflict then `django`, `djangorestframework` these dependencies would not be there. Keep `edx-ace` in the `base.in` as same as `django`. :)

Comment (arjunsinghy96) [reply requirements/dev.txt:24]: Thanks for the update @shimulch . I have moved `edx-ace` to `base.in`.

**Review by shimulch — APPROVED** [review]
@arjunsinghy96 Thanks a lot for the work here. LGTM :+1:

- [x] I tested this locally. Tested if accepting or rejecting verification request sends email notification to the student.
- [x] I read through the code
- [N/A] I checked for accessibility issues
- [x] Includes documentation
- [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Pull request merged**